cfme/utils/net.py

```diff
@@ -58,6 +60,39 @@ def ip_echo_socket(port=32123):
             conn.close()
 
 
+def _trying_fresh_ips(vm, delay, rounds=-1):
+    """ Iterates over IPs from vm for which connection attempts will be made.
+    After all all IPs tried out, this will sleep a while and return next round.
+
+    If rounds is -1, then keep generating ips forever.
+    """
+    _round = 0
+    while _round < rounds or rounds == -1:
+        _round += 1
+        for ip in vm.all_ips:
+            yield ip
+        time.sleep(delay)
+
+
+def pick_responding_ip(vm, port, rounds, rounds_delay_seconds, attempt_timeout):
```

Comment:
Optional, future addition - 

I think we can likely default this to 22, assuming we're looking for something that is responding for SSH.

Reply:
I would like it to stay as it is. While I see this function could be used for finding IP connective with SSH in the future quite a lot, It already is used for finding IPs responding on other than SSH ports (NFS and CIFS). It is meant to be generic. For same reason it is put in `cfme.utils.net`.